src/operators.jl

```diff
@@ -74,27 +75,12 @@ Edge 4 => 5
 Edge 5 => 4
 ```
 """
-function reverse end
-@traitfn function reverse(g::G::IsDirected) where G<:AbstractSimpleGraph
-    gnv = nv(g)
-    gne = ne(g)
-    h = SimpleDiGraph(gnv)
-    h.fadjlist = deepcopy_adjlist(g.badjlist)
-    h.badjlist = deepcopy_adjlist(g.fadjlist)
-    h.ne = gne
-    return h
-end
-
-"""
-    reverse!(g)
-
-In-place reverse of a directed graph (modifies the original graph).
-See [`reverse`](@ref) for a non-modifying version.
-"""
-function reverse! end
-@traitfn function reverse!(g::G::IsDirected) where G<:AbstractSimpleGraph
-    g.fadjlist, g.badjlist = g.badjlist, g.fadjlist
-    return g
+function reverse(g::T) where {T<:AbstractGraph}
+    r = T(nv(g))
```

Comment:
Would suggest to use `G` or `AG` for the type of the graph, if have have some consistency there, it makes it easier to quickly parse the code.

Reply:
Agreed. This was just a code copy, but I'll make the change.

Reply:
Note that `T` representing an AbstractGraph is common throughout the codebase, though.